cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1410,3 +1474,42 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 
 	return nil
 }
+
+func finalizeTryRecoverySystemAndReboot(outcome boot.TryRecoverySystemOutcome) (err error) {
+	// from this point on, we must finish with a system reboot
+	defer func() {
+		if rebootErr := boot.InitramfsReboot(); rebootErr != nil {
+			if err != nil {
+				err = fmt.Errorf("%v (cannot reboot to run system: %v)", err, rebootErr)
+			} else {
+				err = fmt.Errorf("cannot reboot to run system: %v", rebootErr)
+			}
+		}
+	}()
+
+	healthCheck := func() error {
+		// check that writable is accessible by checking whether the
+		// state file exists
+		if !osutil.FileExists(dirs.SnapStateFileUnder(boot.InitramfsHostWritableDir)) {
+			return fmt.Errorf("host state file is not accessible")
+		}
+		return nil
+	}
+
+	if outcome == boot.TryRecoverySystemOutcomeSuccess {
+		if err := healthCheck(); err != nil {
```

Comment:
this err is not assigning to the one in the function as intended, should this be

```suggestion
		if err = healthCheck(); err != nil {
```

that way it can be included in the first error message in the defer'd function

Reply:
I'm not sure what's the original intention here, but this change in particular doesn't help surfacing the error because we will still end with "return nil"  and that nil will be assigned to err before we process the defers

Reply:
I've added a log here instead.